common/src/IO/Tokenizer.h

```diff
@@ -21,97 +21,183 @@
 
 #include "Token.h"
 
-#include <memory>
+#include "Exceptions.h"
+#include "Macros.h"
+
+#include <kdl/string_format.h>
+
 #include <string>
 #include <string_view>
 
 namespace TrenchBroom {
     namespace IO {
-        class TokenizerState {
-        private:
+        struct TokenizerState {
+            const char* cur;
+            size_t line;
+            size_t column;
+            bool escaped;
+        };
+
+        struct TokenizerStateAndSource {
+            TokenizerState state;
+            const char* begin;
+            const char* end;
+        };
+
+        class TokenizerBase {
+        protected:
             const char* m_begin;
-            const char* m_cur;
             const char* m_end;
             std::string m_escapableChars;
             char m_escapeChar;
-            size_t m_line;
-            size_t m_column;
-            bool m_escaped;
+            TokenizerState m_state;
         public:
-            TokenizerState(const char* begin, const char* end, const std::string& escapableChars, char escapeChar);
+            TokenizerBase(const char* begin, const char* end, std::string_view escapableChars, char escapeChar);
+
+            inline void replaceState(std::string_view str) {
```

Comment:
Did you take any measurements? I'd be interested to see if any (and if so, which) of these inlined functions actually make a difference in runtime.

Reply:
Yes, profiled on MSVC and Clang/macOS. Getting the compiler to inline everything is the most important change here, performance wise.
I can give some updated numbers when I'm done the other changes

Reply:
That's interesting, I wouldn't have expected that. Yeah, I'd be interested in seeing the numbers!